localstack/services/apigateway/apigateway_listener.py

```diff
@@ -1041,7 +1041,9 @@ def invoke_rest_api_integration_backend(invocation_context: ApiInvocationContext
 
 def get_target_resource_details(invocation_context: ApiInvocationContext) -> Tuple[str, Dict]:
     """Look up and return the API GW resource (path pattern + resource dict) for the given invocation context."""
-    path_map = helpers.get_rest_api_paths(rest_api_id=invocation_context.api_id)
+    path_map = helpers.get_rest_api_paths(
```

Comment:
This change looks good, as it solves the current issue with selecting the right region. 👍 

General observation: looking at the implementation of `get_rest_api_paths(..)`, I feel like we could consider doing an in-memory lookup of the target moto RestAPI model instance directly (rather than using `aws_stack.connect_to_service(..)` and looping over all `resources` and doing another HTTP call for each of them). This way, we could potentially also get rid of the `region_name` parameter here (as we can simply iterate all regions in-memory). Thoughts?

Reply:
Sure, that's an interesting discussion alright. There are definitely APIs we want to pass the region name as a parameter as part of the business logic, examples are, multi-region tests or multi-region calls. I don't this is one of them.

Now, you are spot on regarding lookups we do aftermath to obtain some resources from an API. I can see a solution like this:

1. Wrap that lookup logic for common operations around an `ApiLookup...` class that can have those functions. You can see that scattered in the code to lookup integrations, resources etc. We can even make it smarter by having some caching.
2. After these calculations, use ApiInvocationContext to hold that state.

My main issue is not with this method (mainly) is using that lookup approach in all the places, but only from an efficiency perspective but mainly because it doesn't remove these little functions out of the main logic.